### ISSUE TITLE
CtcLossLayer does not crash when logits and targets are not matching

### ISSUE THREAD

**Issue by JackTemaki** [rwth-i6/returnn#1131]
I accidentally constructed a training run where the number of targets was higher than the number of logits passed to a `CtcLossLayer`, and still the training finished without error. This is not the behavior I expected, and I do not understand how the code can work without crashing.

**Comment by albertz** [comment]
Ok, I added a check.
